front/src/connections/Connection.ts

```diff
@@ -0,0 +1,32 @@
+import { ReceiverBase } from '../controllers/ReceiverBase';
+import { Emitter } from '../controllers/Emitter';
+
+export abstract class Connection {
+    private receivers: ReceiverBase[] = [];
+
+    // データ送信用
+    abstract emitData(type: string, data: any): void;
+
+    // レシーバーを登録する
+    public setReceivers(receiver: ReceiverBase) {
+        this.receivers.push(receiver);
+    }
+
+    // エミッターを作成する TODO: もう少しいい感じの設計をしたい
+    public createEmitter() {
+        return new Emitter((type: string, data?: any) => {
```

Comment:
dataはundefinableよりもなかったらthrowする方がいいかなと思ったけど，何もPayloadがないエンドポイントを叩くようにundefinedになっているという認識であってる？

であれば，

```suggestion
        return new Emitter((type: string, data: any = {}) => {
```